compiler/src/dmd/expression.d

```diff
@@ -249,18 +249,54 @@ bool isDotOpDispatch(Expression e)
 }
 
 /****************************************
- * Expand tuples.
- * Input:
- *      exps    aray of Expressions
- * Output:
- *      exps    rewritten in place
+ * Expand tuples in-place.
+ *
+ * Params:
+ *     exps  = array of Expressions
+ *     names = optional array of names corresponding to Expressions
  */
-extern (C++) void expandTuples(Expressions* exps)
+extern (C++) void expandTuples(Expressions* exps, Identifiers* names = null)
 {
     //printf("expandTuples()\n");
     if (exps is null)
         return;
 
+    if (names)
+    {
+        if (exps.length != names.length)
+        {
+            // Fixme: this branch should be removed when bugs are fixed.
```

Comment:
What bugs are you referring to?

Reply:
Because the name is no longer stored in an expression node, any modifications to CallExp.arguments (think UFCS rewrites, operator overloading) may desync the names and arguments. The rewrites for functions will be fixed to account for that in the next PR.